src/containerapp/azext_containerapp/_validators.py

```diff
@@ -15,13 +15,29 @@
 from ._utils import safe_get, is_registry_msi_system
 from ._constants import ACR_IMAGE_SUFFIX, LOG_TYPE_SYSTEM, CONNECTED_ENVIRONMENT_RESOURCE_TYPE, \
     CONNECTED_ENVIRONMENT_TYPE, MANAGED_ENVIRONMENT_RESOURCE_TYPE, MANAGED_ENVIRONMENT_TYPE, CONTAINER_APPS_RP, \
-    EXTENDED_LOCATION_RP, CUSTOM_LOCATION_RESOURCE_TYPE
+    EXTENDED_LOCATION_RP, CUSTOM_LOCATION_RESOURCE_TYPE, MAXIMUM_SECRET_LENGTH
+from urllib.parse import urlparse
 
 logger = get_logger(__name__)
 
 
 # called directly from custom method bc otherwise it disrupts the --environment auto RID functionality
-def validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait):
+def validate_create(registry_identity, registry_pass, registry_user, registry_server, no_wait, source=None, repo=None, yaml=None, environment_type=None):
+    if source and repo:
+        raise MutuallyExclusiveArgumentError("Usage error: --source and --repo cannot be used together. Can either deploy from a local directory or a GitHub repository")
+    if (source or repo) and yaml:
+        raise MutuallyExclusiveArgumentError("Usage error: --source or --repo cannot be used with --yaml together. Can either deploy from a local directory or provide a yaml file")
+    if (source or repo) and environment_type == CONNECTED_ENVIRONMENT_TYPE:
+        raise MutuallyExclusiveArgumentError("Usage error: --source or --repo cannot be used with --environment-type connectedEnvironment together. Please use --environment-type managedEnvironment")
+    if source or repo:
+        if not registry_server:
+            raise RequiredArgumentMissingError('Usage error: --registry-server is required while using --source or --repo')
+    if repo and registry_server and "azurecr.io" in registry_server:
+        parsed = urlparse(registry_server)
+        registry_name = (parsed.netloc if parsed.scheme else parsed.path).split(".")[0]
+        if registry_name and len(registry_name) > MAXIMUM_SECRET_LENGTH:
+            raise ValidationError(f"--registry-server ACR name must be less than {MAXIMUM_SECRET_LENGTH} "
```

Comment:
```suggestion
            raise InvalidArgumentValueError(f"--registry-server ACR name must be less than {MAXIMUM_SECRET_LENGTH} "
```

Reply:
Actually, `ValidationError` is acceptable, but personally, I think `InvalidArgumentValueError` is a more accurate usage.
If you want to maintain consistency with `_validate_up_args`, then I think it's also OK. Or modify them all to `InvalidArgumentValueError`

Reply:
Thank you for taking a look! Just going to keep it as-is so we are consistent with [this validation](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/_up_utils.py#L770) in the `up` command.

Reply:
OK